rust/xaynet-server/src/state_machine/phases/error.rs

```diff
@@ -25,6 +25,9 @@ pub enum PhaseStateError {
     Timeout(#[from] tokio::time::Elapsed),
     #[error("redis request failed: {0}")]
     Redis(#[from] RedisError),
+    #[cfg(feature = "model-persistence")]
+    #[error("s3 request failed: {0}")]
+    S3(#[from] crate::storage::s3::S3Error),
```

Comment:
Do we actually use the inner error? If not I think we don't have to store the inner error. We could have `SaveGlobalModel` and `LoadGlobalModel` variants instead, which seem more specific to the phases logic.

Reply:
Yes that is a good idea👍

Reply:
I tried to implement it but not sure how to implement a quick good solution here. I think it is still good to keep the inner error so that we can log the s3 error here

https://github.com/xaynetwork/xaynet/blob/7f5dc2eb65787654a9dfa0be5f734702b0684dd1/rust/xaynet-server/src/state_machine/phases/error.rs#L45

Otherwise I would need to log the error here:
https://github.com/xaynetwork/xaynet/blob/2d6cededc16a95be503cec4a8a4f7f7db53fea30/rust/xaynet-server/src/state_machine/phases/unmask.rs#L77
or in the s3 client. 

But I like the idea to hide the s3 and redis errors behind more higher level error variants.

Is it fine if we pospone it and maybe combine it with the ideas that I wrote in #544 (defining traits and using `tower`)?